components/movementsensor/gpsutils/vrs.go

```diff
@@ -66,3 +82,50 @@ func HasVRSStream(sourceTable *Sourcetable, mountPoint string) (bool, error) {
 
 	return stream.Nmea, nil
 }
+
+// Close closes the VRS connection and any other background threads.
+func (vrs *VRS) Close() error {
+	vrs.cancelFunc()
+	vrs.activeBackgroundWorkers.Wait()
+	return vrs.Conn.Close()
+}
+
+// StartGGAThread starts a thread that writes GGA messages to the VRS.
+func (vrs *VRS) StartGGAThread(ggaFunc func() (string, error)) {
+	vrs.activeBackgroundWorkers.Add(1)
+	go func() {
+		defer vrs.activeBackgroundWorkers.Done()
+		ticker := time.NewTicker(time.Duration(1000.*20) * time.Millisecond)
```

Comment:
You're multiplying by 1000 and then using `Time.Millisecond`. I suspect this should be `time.Duration(20 * time.Second)`

Reply:
OH I remembered the reason for this, basically time.Duration has to be an integer, and usually when I do this I need values that are smaller than a second. this code def doesn't need millisecond values